File: 2025/R/solve_day1_part1.R
solve_day1_part1 <- function(input) {
  pos <- 50L
  dial_size <- 100L
  zero_counter <- 0L

  dir <- substr(input, 1L, 1L)
  val <- as.integer(substr(input, 2L, nchar(input)))

  turns <- ifelse(dir == "R", val, -val)

  for (turn in turns) {
    pos <- (pos + turn) %% dial_size

    zero_counter <- (pos == 0L) + zero_counter
  }

  zero_counter
}

File: 2025/python/solve_day1_part1.py
def solve_day1_part1(text):
    zero_counter = 0
    dial_size = 100
    pos = 50

    turns = [turn.replace("L", "-").replace("R", "") for turn in text]
    turns = [int(turn) for turn in turns]

    for turn in turns:
        pos = (pos + turn) % dial_size

        zero_counter += pos == 0

    return(zero_counter)
